test/Npgsql.Tests/YBTopologyAwareRRSupportTests.cs

```diff
@@ -0,0 +1,989 @@
+
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine("Failure:" + ex.Message);
+            Console.WriteLine("Failure stacktrace: " + ex.StackTrace);
+        }
+        finally
+        {
+            foreach (var conn in conns)
+            {
+                conn.Close();
+            }
+            VerifyLocal("127.0.0.1", 0);
+            VerifyLocal("127.0.0.2", 0);
+            VerifyLocal("127.0.0.3", 0);
+            VerifyLocal("127.0.0.4", 0);
+            VerifyLocal("127.0.0.5", 0);
+            VerifyLocal("127.0.0.6", 0);
+            DestroyCluster();
+        }
+    }
+
+    static List<NpgsqlConnection> CreateConnections(string connString, int numConns)
+    {
+        List<NpgsqlConnection> conns = new List<NpgsqlConnection>();
+        try
+        {
+            for (var i = 1; i <= numConns; i++)
+            {
+                NpgsqlConnection conn = new NpgsqlConnection(connString);
+                conn.Open();
+                conns.Add(conn);
+            }
+
+            Console.WriteLine("Connections Created");
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine("Failure:" + ex.Message);
+            Console.WriteLine("Failure stacktrace: " + ex.StackTrace);
+            return conns;
+        }
+
+        return conns;
+
+    }
+
+    void CreateRRCluster()
+    {
+        string? _Output = null;
+        string? _Error = null;
+        var cmd = "/bin/yb-ctl create --rf 3 --placement_info cloud1.datacenter1.rack1,cloud1.datacenter2.rack1,cloud1.datacenter3.rack1 --tserver_flags \"placement_uuid=live,max_stale_read_bound_time_ms=60000000\"";
+        ExecuteShellCommand(cmd, ref _Output, ref _Error );
+        Console.WriteLine("Output:" + _Output);
+        cmd = "/build/latest/bin/yb-admin --master_addresses 127.0.0.1:7100,127.0.0.2:7100,127.0.0.3:7100 modify_placement_info cloud1.datacenter1.rack1,cloud1.datacenter2.rack1,cloud1.datacenter3.rack1 3 live";
+        ExecuteShellCommand(cmd, ref _Output, ref _Error );
+        Console.WriteLine("Output:" + _Output);
+        cmd = "/bin/yb-ctl add_node --placement_info cloud1.datacenter2.rack1 --tserver_flags placement_uuid=rr";
+        ExecuteShellCommand(cmd, ref _Output, ref _Error );
+        Console.WriteLine("Output:" + _Output);
+        cmd = "/bin/yb-ctl add_node--placement_info cloud1.datacenter3.rack1 --tserver_flags placement_uuid=rr";
+        ExecuteShellCommand(cmd, ref _Output, ref _Error );
+        Console.WriteLine("Output:" + _Output);
+        cmd = "/bin/yb-ctl add_node --placement_info cloud1.datacenter4.rack1 --tserver_flags placement_uuid=rr";
+        ExecuteShellCommand(cmd, ref _Output, ref _Error );
+        Console.WriteLine("Output:" + _Output);
+    }
+
+    protected void DestroyCluster()
+    {
+        string? _Output = null;
+        string? _Error = null;
+        var cmd = "/bin/yb-ctl destroy";
+        ExecuteShellCommand(cmd, ref _Output, ref _Error );
+    }
+}
```

Comment:
Please look into moving common code in to a separate method that can be called from wherever needed.

Reply:
Done